### PR TITLE
DCJ-668: Bug Fixes for Terms of Service Acceptance

### PR DESCRIPTION
**Pull request by rushtong**
### Addresses
https://broadworkbench.atlassian.net/browse/DCJ-668

### Summary
https://github.com/DataBiosphere/duos-ui/pull/2657 introduced a couple of regressions that this PR fixes.

1. Don't sign out when oidc user is not present - it won't be until we have oidc sign-in
2. Don't sign out if the user has authenticated, but hasn't accepted ToS
3. Re-order the tos checks to help simplify the process
4. Lint the Sign-in component

Suggested that you hide whitespace changes due to the large number of lint fixes.

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/components/SignInButton.tsx [hunk 1]
```diff
@@ -118,13 +128,6 @@ export const SignInButton = (props: SignInButtonProps) => {
 
   const shouldRedirectTo = (page: string): boolean => page !== '/' && page !== '/home';
 
-  const attemptSignInCheckToSAndRedirect = async (redirectTo:string, shouldRedirect: boolean) => {
-    await checkToSAndRedirect(shouldRedirect ? redirectTo : null);
```
Comment (rushtong) [inline src/components/SignInButton.tsx:122]: I moved this line to `onSuccess` to simplify the logic checks

File: src/components/SignInButton.tsx [hunk 2]
```diff
@@ -118,13 +128,6 @@ export const SignInButton = (props: SignInButtonProps) => {
...
-    Metrics.identify(Storage.getAnonymousId());
-    Metrics.syncProfile();
-    Metrics.captureEvent(eventList.userSignIn);
```
Comment (rushtong) [inline src/components/SignInButton.tsx:125]: I moved these to `checkToSAndRedirect` in the place where we know the user is authenticated and accepted ToS.

File: src/components/SignInButton.tsx [hunk 3]
```diff
@@ -73,22 +77,28 @@ export const SignInButton = (props: SignInButtonProps) => {
...
     }
     setUserRoleStatuses(user, Storage);
     await onSignIn();
     const userStatus = await ToS.getStatus();
-    const { tosAccepted } = userStatus;
+    const {tosAccepted} = userStatus;
     if (!isEmpty(userStatus) && !tosAccepted) {
-      await Auth.signOut();
+      // User has authenticated, but has not accepted ToS
+      Storage.setUserIsLogged(false);
```
Comment (rushtong) [inline src/components/SignInButton.tsx:90]: This is the fix that allows for an authenticated user to access `/tos_acceptance`

File: src/libs/auth/auth.ts [hunk 4]
```diff
@@ -24,11 +23,6 @@ export const Auth = {
       Auth.signOut();
       //TODO: DUOS-3082 Add an alert that session has expired
     });
-    if (oidcUser !== null) {
-      Storage.setUserIsLogged(true);
-    } else {
-      await Auth.signOut();
-    }
```
Comment (rushtong) [inline src/libs/auth/auth.ts:31]: This fix allows for page refreshes without signing out the user.

**Review by fboulnois — APPROVED** [review]
looks good 👍

**Review by rjohanek — APPROVED** [review]
changes look good

**Review by s-rubenstein — APPROVED** [review]

**Pull request merged**